### PR TITLE
samples: peripheral_fast_pair: Limit discoverable adv session

### PR DESCRIPTION
**Pull request by MarekPieta**
Change limits time of discoverable advertising session to fulfill Fast Pair requirements.

Jira: NCSDK-14830

### CODE REVIEW

File: samples/bluetooth/peripheral_fast_pair/README.rst [hunk 1]
```diff
@@ -94,6 +94,7 @@ Button 1:
 
    .. note::
        The advertising is disabled while the Fast Pair Provider is connected to a Bluetooth Central.
+       The discoverable advertising is automatically switched to not discoverable advertising with show UI indication after a 10-minute timeout.
```
Comment (divipillai) [inline samples/bluetooth/peripheral_fast_pair/README.rst:97]:
```suggestion
       The discoverable advertising is automatically switched to not discoverable advertising with the show UI indication mode after a 10-minute timeout.
```

File: samples/bluetooth/peripheral_fast_pair/src/main.c [hunk 2]
```diff
@@ -200,13 +230,12 @@ static void volume_control_btn_handle(uint32_t button_state, uint32_t has_change
 
 static void fp_adv_mode_btn_handle(uint32_t button_state, uint32_t has_changed)
 {
-	struct k_work_sync sync;
 	uint32_t button_pressed = button_state & has_changed;
 
 	if (button_pressed & FP_ADV_MODE_BUTTON_MASK) {
-		k_work_cancel_delayable_sync(&fp_adv_mode_status_led_handle, &sync);
+		k_work_cancel_delayable(&fp_adv_mode_status_led_handle);
```
Comment (alstrzebonski) [inline samples/bluetooth/peripheral_fast_pair/src/main.c:236]: This line can be removed. We later reschedule this work anyway and we are in cooperative thread.

**Review by alstrzebonski — CHANGES_REQUESTED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 3]
```diff
@@ -190,7 +190,10 @@ Bluetooth samples
 
 * :ref:`peripheral_fast_pair` sample:
 
-  * Added possibility of toggling between show and hide UI indication in the Fast Pair not discoverable advertising.
+  * Added:
+
+    * Possibility of toggling between show and hide UI indication in the Fast Pair not discoverable advertising.
+    * Automatic switching to not discoverable advertising with the show UI indication mode after 10 minutes of discoverable advertising.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:196]:
```suggestion
    * Automatic switching to the not discoverable advertising with the show UI indication mode after 10 minutes of discoverable advertising.
```

File: samples/bluetooth/peripheral_fast_pair/README.rst [hunk 4]
```diff
@@ -94,6 +94,7 @@ Button 1:
 
    .. note::
        The advertising is disabled while the Fast Pair Provider is connected to a Bluetooth Central.
+       The discoverable advertising is automatically switched to not discoverable advertising with the show UI indication mode after a 10-minute timeout.
```
Comment (greg-fer) [inline samples/bluetooth/peripheral_fast_pair/README.rst:97]:
```suggestion
       The discoverable advertising is automatically switched to the not discoverable advertising with the show UI indication mode after a 10-minute timeout.
```

**Review by greg-fer — CHANGES_REQUESTED** [review]

**Review by greg-fer — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]

**Pull request merged**